### ISSUE TITLE
Mismatching number of %-placeholders in string

### ISSUE THREAD

**Issue by kaddkaka** [astral-sh/ruff#24164]
### Summary

Like #24031.

Incorrect number of format specifies should give error.

There are already many good warnings and tips about using f-strings. Howerever I believe this code does not give any warning about number of arguments:
```py
"" % banana
```

Since there is 0 placeholders in the string, there should be no reason to use `%` at all. 

Cases to solve for(?) :
```
"hello" % x  # string literal
f"world" % y  # f-string? (maybe not because f-string interpolation could produce placeholders) 
"hello " * 3 % z  # can this be solved for free? 
```

Any more cases that are possible to solve? Easily? 

Notes: x could be a tuple with 0 elements, but I can't see any meaningful value in code like that. But does that fact mean this should be it's own lint rule separate from F507? 

Related:
- #24031
- #24142

**Comment by MichaReiser** [comment]
Extending F507 to also catch `"" % banana` where banana is known to not be a zero length tuple (use Ruff's type inference) does make sense to me. 

> f"world" % y  # f-string? (maybe not because f-string interpolation could produce placeholders) 
"hello " * 3 % z  # can this be solved for free? 

Let's ignore f-strings and multiplications in this issue.

**Comment by kaddkaka** [comment]
Could we also catch `"" % banana` even if banana has unknown value? Or should that be a new separate rule that says: "usage of `%` with string without placeholders"?

**Comment by MichaReiser** [comment]
> Could we also catch "" % banana even if banana has unknown value? 

I guess we could. The chance where this is intentional is very low

**Comment by spider-yamet** [comment]
@MichaReiser May I work on this issue? then, please feel free the assign this for me.
Thanks

**Comment by spider-yamet** [comment]
I think F507 is designed for runtime-mismatch checker, but for this issue, we have to catch weird style pattern as well. 
Like in "hello" % banana, banana is unknown at lint time, so it is skipped to avoid false positives.
So the problem is rule mismatch, I think. Would appreciate your feedback, Thanks

**Comment by spider-yamet** [comment]
Please let me know whether I can open a PR for this, @MichaReiser Thanks
